components/logins/src/store.rs

```diff
@@ -24,15 +24,25 @@ pub fn get_registered_sync_engine(engine_id: &SyncEngineId) -> Option<Box<dyn Sy
     let weak = STORE_FOR_MANAGER.lock();
     match weak.upgrade() {
         None => None,
-        Some(store) => match engine_id {
-            SyncEngineId::Passwords => Some(Box::new(LoginsSyncEngine::new(Arc::clone(&store)))),
-            // panicing here seems reasonable - it's a static error if this
-            // it hit, not something that runtime conditions can influence.
-            _ => unreachable!("can't provide unknown engine: {}", engine_id),
+        Some(store) => match create_sync_engine(store, engine_id) {
+            Ok(engine) => Some(engine),
+            Err(e) => {
+                log::error!("logins: get_registered_sync_engine: {}", e);
+                None
+            }
         },
     }
 }
 
+fn create_sync_engine(store: Arc<LoginStore>, engine_id: &SyncEngineId) -> Result<Box<dyn SyncEngine>> {
+    match engine_id {
+        SyncEngineId::Passwords => Ok(Box::new(LoginsSyncEngine::new(Arc::clone(&store))?)),
+        // panicing here seems reasonable - it's a static error if this
+        // it hit, not something that runtime conditions can influence.
+        _ => unreachable!("can't provide unknown engine: {}", engine_id),
+    }
+}
```

Comment:
It's a bit annoying that we have to refactor the code like this to deal with the fact that creating a new interrupt scope can now fail, therefore creating a new sync engine can also fail.  However, this does deal with the case where we initiate shutdown mode, then another thread tries to start a sync.